src/test/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSinkTest.java

```diff
@@ -86,6 +86,32 @@ public void testInjectExcludeWithout() {
         Mockito.verify(_target).recordAggregateData(dataIncluded);
     }
 
+    @Test
+    public void testWhistlist() {
+        final DimensionFilteringSink sink = new DimensionFilteringSink.Builder()
+                .setName("testWhistlist")
+                .setSink(_target)
+                .setWhitelistDimensions(ImmutableSet.of("foo", "bar"))
+                .build();
+        final PeriodicData dataExcluded = TestBeanFactory.createPeriodicDataBuilder()
+                .setDimensions(ImmutableMap.of("foo", "fooVal", "baz", "bazVal"))
+                .build();
+        sink.recordAggregateData(dataExcluded);
+        Mockito.verify(_target, Mockito.never()).recordAggregateData(Mockito.any(PeriodicData.class));
+
+        final PeriodicData dataIncluded = TestBeanFactory.createPeriodicDataBuilder()
```

Comment:
The code does this:
if not (whitelist contains all of the metrics dimensions) { exclude the metric }

For example, if the whitelist is `[a, b, c]`, and the dimensions of the metric are `[a, b]` then (whitelist contains all of the metrics dimensions) is true and the metric is allowed through. 

For the same whitelist, if the dimensions of the metric are `[a, b, g]` then (whitelist contains all of the metrics dimensions) is false and the metric is dropped.

Thus, the sink does indeed observe additional dimensions keys and drops the metric if any keys are not on the whitelist.

Super sorry if I'm still mis-understanding!!